netty/src/main/java/io/grpc/netty/JettyTlsUtil.java

```diff
@@ -16,10 +16,39 @@
 
 package io.grpc.netty;
 
+import java.lang.reflect.Method;
+import java.security.AccessController;
+import java.security.PrivilegedExceptionAction;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+
 /**
  * Utility class for determining support for Jetty TLS ALPN/NPN.
  */
 final class JettyTlsUtil {
+  private static final boolean HAS_JAVA9_ALPN;
+
+  static {
+    Method getApplicationProtocol;
+    try {
+      SSLContext context = SSLContext.getInstance("TLS");
+      context.init(null, null, null);
+      SSLEngine engine = context.createSSLEngine();
+      getApplicationProtocol =
+          AccessController.doPrivileged(new PrivilegedExceptionAction<Method>() {
+            @Override
+            public Method run() throws Exception {
+              return SSLEngine.class.getMethod("getApplicationProtocol");
+            }
+          });
+      getApplicationProtocol.invoke(engine);
+    } catch (Throwable t) {
```

Comment:
Could we just catch Exception, or similar? I'd rather not catch every Error and throw it away, as it can hide legitimate problems.